streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -107,18 +84,23 @@ public void addChild(final ProcessorNode<?, ?> child) {
     }
 
     public void init(final InternalProcessorContext context) {
-        this.context = context;
         try {
             nodeMetrics = new NodeMetrics(context.metrics(), name, context);
-            runAndMeasureLatency(time, initDelegate, nodeMetrics.nodeCreationSensor);
+            final long startNs = time.nanoseconds();
+            if (processor != null) {
+                processor.init(context);
+            }
+            nodeMetrics.nodeCreationSensor.record(time.nanoseconds() - startNs);
         } catch (final Exception e) {
             throw new StreamsException(String.format("failed to initialize processor %s", name), e);
         }
     }
 
     public void close() {
         try {
-            runAndMeasureLatency(time, closeDelegate, nodeMetrics.nodeDestructionSensor);
+            if (processor != null) {
+                processor.close();
+            }
```

Comment:
By my reckoning, we are trying to record the "destroy" time as a latency metric, but line 104 unloads the sensor and all the related metrics immediately after recording the destroy latency.

I think this means that the destroy metric has never actually been observable (unless you're super lucky to sample it concurrently between the record and remove.

In that case, we apparently don't need it, so I removed it. Alternatively, we can keep it and not call `nodeMetrics.removeAllSensors()`, or remove all but the destroy sensor...

Or we can leave it as it was, but having nonsense code gives me a funny feeling even when it's harmless.

Of course, please let me know if I've simply mis-understood what was happening here.

Reply:
Interesting find. However I am wondering if we can simply remove a metric? It's a public API change and should require a KIP including deprecation period...

Reply:
Yeah, I'm not sure either. On one hand, it's "dead code" in that it's never updated until it's removed. But on the other hand, it's created when the node is, so it's actually present for all nodes (just always reporting 0).

I can leave it in, since it's harmless, but I wanted to have the discussion, since I'm 99% sure it's also harmless to take it out.